cfme/utils/fileserver.py

```diff
@@ -0,0 +1,190 @@
+
+    The base directory is `miq` and under that, we have directories as per the automation entities;
+    which can be set at initialization.
+
+    Directory Structure:
+    /-miq/
+        /-Database/
+        /-Datastores/
+        /-Dialogs/
+        /-Others/
+        /-Reports/
+
+    Args:
+        entity: entity which you want to access like `Datastores`, `Dialogs`, etc.
+        entrypoint: FTP server entry point
+        host: FTP server host
+        user: FTP user
+        passwd: FTP password
+
+    Usage:
+        .. code-block:: python
+          # Entities
+          fs = FileServer()
+          fs.directories    # list of current available entities
+          fs.mkdir("Dialogs")   # create new entity type
+          fs.rmd("Dialogs")     # delete created entity type
+
+          fs = FileServer("Dialogs")
+          fs.upload("foo.zip") # upload local file
+          fs.files  # return list of available file objects
+          fs.download("foo.zip") # It will download foo.zip file
+
+          f = fs.get_file("foo.zip")
+          f.path  # gives remote file path; can be used to download with 'wget'
+          f.download()  # download file
+    """
+
+    def __init__(self, entity=None, entrypoint=None, host=None, user=None, passwd=None):
+        ftp_data = cfme_data.ftpserver
+
+        self.entity = entity
+        self.host = host if host else ftp_data.host
+        self.user = user if user else credentials[ftp_data.credentials]["username"]
+        self.passwd = passwd if passwd else credentials[ftp_data.credentials]["password"]
+        self.entrypoint = entrypoint or ftp_data.entrypoint
+
+        self.client = FTPClient(self.host, self.user, self.passwd, time_diff=False)
+        self.cwd = os.path.join(self.entrypoint, "miq", self.entity if entity else "")
+        self.client.cwd(self.cwd)
+
+    @property
+    def _entities(self):
+        result = {"files": [], "dirs": []}
+
+        def _checks(line):
+            entity = re.split(r"\s+", line, maxsplit=8)[-1]
+            if line.startswith("d"):
+                result["dirs"].append(entity)
+            else:
+                result["files"].append(entity)
+
+        self.client.ftp.dir(_checks)
+        return result
+
+    @property
+    def files(self):
+        """List of FTP file objects"""
+        return [File(name=f, client=self) for f in self._entities["files"]]
+
+    def get_file(self, name):
+        """
+        Arg:
+            name: name of remote file
+        Returns:
+            FTP file object
+        """
+
+        if name in self._entities["files"]:
+            return File(name=name, client=self)
+        else:
+            raise FTPException("{} not found".format(name))
+
+    def rm(self, name):
+        """ Remove or Delete remote FTP file
+        Arg:
+            name: name of remote file
+        Returns:
+            Success of the action
+        """
+        return self.client.dele(name)
```

Comment:
Required : Change this to return self.client.delete(name)

Reply:
now it will take `dele` only as we are making a wrapper